### PR TITLE
feat: Prisma Server to watch a node

### PR DESCRIPTION
**Pull request by yjkimjunior**
I was thinking to store everything as `Codec` but Prisma doesn't yet support the Postgresql `bytea` https://github.com/prisma/prisma/issues/2466

so I guess the workaround options are:
1. decode Codec to String or Int and store it, which would be quite bad
2. serialize the Codec and store it as a [String!]!, which is less bad

Main things are:
* DB Schema: as discussed in https://github.com/yjkimjunior/Nomidot/issues/10 in `datamodel.prisma`
* Docker Images: currently just local until we get access to hosting
* Prisma Client: provides GraphQL API to access our DB, auto generated according to `datamodel.prisma`
* Tasks: anyone who wants to use this node watcher to read/write their own models to their own db can just write a `Task` that describes how to do that.

### CODE REVIEW

File: back/node_watcher/src/prismaWriter.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  constructor() {
+    let queue = this.queue;
+  }
+
+  public addToQueue(item: any): void { // FIXME any
+    this.queue.push(item);
+  }
+
+  private async writeBlockNumber({ block_number, authored_by, block_hash }: BlockInfoFromNode) {
```
Comment (amaurymartiny) [inline back/node_watcher/src/prismaWriter.ts]: Use camel case everywhere since it's JS standard

File: back/node_watcher/src/prismaWriter.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { prisma } from '../generated/prisma-client';
+import { BlockInfoFromNode } from './types';
+
+/*
+* Handles async/await writing to PSQL with Prisma GraphQL client
+* Basically subscriptions to node could move on before async write resolves
+* so we put them in a queue here.
```
Comment (amaurymartiny) [inline back/node_watcher/src/prismaWriter.ts]: I would be way more easier to remove the queue, never subscribe to tip, and use `.at` everywhere

Comment (yjkimjunior) [reply back/node_watcher/src/prismaWriter.ts]: ah, forgot about `.at`.

File: back/node_watcher/src/node-watcher.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  private prismaClient: PrismaClient;
+
+  constructor(api: ApiPromise, options: NodeWatcherOptions) {
+    this.nodeApiInstance = new NodeApi(api, options);
+    this.prismaClient = new PrismaClient();
+  }
+
+  public start() {
+    from(BLOCK_NUMBER_RANGE) // fixme: make range to max safe integer
+      .subscribe((number: number) => {
```
[the remark this answers is not in the document]
Comment (amaurymartiny) [reply back/node_watcher/src/node-watcher.ts]: Also, let's remove rxjs. I would prefer either to go 100% rx, or go 100% promise.

File: back/node_watcher/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import NodeApi from './node-api';
+import NodeWatcher from './node-watcher';
+import { NodeWatcherOptions } from './types';
+
+const ENDPOINT = 'ws://127.0.0.1:9944';
+
+async function main () {
+  const provider = new WsProvider(ENDPOINT);
+  const api = await new ApiPromise({ provider });
```
[the remark this answers is not in the document]
Comment (yjkimjunior) [reply back/node_watcher/src/index.ts]: Weird, that's how it's documented and tested throughout polkadotjs though.

e.g. https://github.com/polkadot-js/api/blob/cc432e2291ee5591fcb788c54301f49b3a57d84c/packages/api/test/e2e/api/promise-tx-eras.spec.ts#L27

[the remark this answers is not in the document]
Comment (yjkimjunior) [reply back/node_watcher/src/index.ts]: https://github.com/polkadot-js/api/blob/cc432e2291ee5591fcb788c54301f49b3a57d84c/packages/api/test/e2e/api/promise-tx-eras.spec.ts#L27

[the remark this answers is not in the document]
Comment (yjkimjunior) [reply back/node_watcher/src/index.ts]: Weird, that's how ApiPromise gets initialized everywhere AFAICT, 

e.g. https://github.com/polkadot-js/api/blob/cc432e2291ee5591fcb788c54301f49b3a57d84c/packages/api/test/e2e/api/promise-tx-eras.spec.ts#L27

[the remark this answers is not in the document]
Comment (amaurymartiny) [reply back/node_watcher/src/index.ts]: `ApiPromise.create` is basically a wrapper around `new ApiPromise() + api.isReady`

**Pull request merged**